adaptdl/adaptdl/checkpoint.py

```diff
@@ -130,10 +163,15 @@ def load_state(state):
     """
     if checkpoint_path() is None:
         return False
-    try:
-        name = _STATES_TO_NAMES[state]
-        with open(os.path.join(checkpoint_path(), name), "rb") as f:
-            state.load(f)
-        return True
-    except FileNotFoundError:
+
+    name = _STATES_TO_NAMES[state]
+    ckpt_dir = os.path.join(checkpoint_path(),
+                            f"{CKPT_DIR_PREFIX}{num_restarts() - 1}")
```

Comment:
This makes it if the previous checkpoint failed, it will restart from the beginning. Probably better to find the latest checkpoint, even if it's not the immediate previous restart. Can output a warning if it's not from the previous restart since that can be useful debugging info.

Reply:
Good catch. I have updated the code. I will do a quick integration test with resnet.

Reply:
Ran `resnet18-cifar10-elastic.sh` and the result looks good.